src/finch/finchassembly/interpreter.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import dataclass
+from itertools import product
+from typing import Iterable, Any
+import numpy as np
+from ..finch_logic import (
+    Immediate,
+    Deferred,
+    Field,
+    Alias,
+    Table,
+    MapJoin,
+    Aggregate,
+    Query,
+    Plan,
+    Produces,
+    Subquery,
+    Relabel,
+    Reorder,
+)
+from ..algebra import return_type, fill_value, element_type, fixpoint_type
+
+
+@dataclass(eq=True, frozen=True)
+class TableValue:
+    tns: Any
+    idxs: Iterable[Any]
+
+    def __post_init__(self):
+        if isinstance(self.tns, TableValue):
+            raise ValueError("The tensor (tns) cannot be a TableValue")
+
+
+class FinchLogicInterpreter:
```

Comment:
This seems like a duplicate of https://github.com/finch-tensor/finch-tensor-lite/blob/c0cf9908abd0affae0d81581c54fe32f05065854/src/finch/finch_logic/interpreter.py#L36

Reply:
Yes, I should have deleted that when I copied the classes.